src/test/java/com/connexta/search/SearchITest.java

```diff
@@ -97,309 +90,485 @@ public URL solrUrl() throws MalformedURLException {
+
+  @Inject private WebTestClient webTestClient;
+
+  @Value("${endpoints.index.version}")
+  private String indexApiVersion;
+
+  private static MockWebServer storeMockWebServer;
+
+  @BeforeEach
+  void beforeEach() throws IOException, SolrServerException {
+    storeMockWebServer = new MockWebServer();
+    storeMockWebServer.start();
+
     // TODO shouldn't need to clear solr every time
     solrClient.deleteByQuery(SOLR_COLLECTION, "*");
     solrClient.commit(SOLR_COLLECTION);
   }
 
-  @ParameterizedTest
-  @ValueSource(strings = {" ", "text"})
-  public void testStoringValidCst(String contents) throws Exception {
-    // given
-    final String indexEndpointUrl = INDEX_ENDPOINT_BASE_URL + "00067360b70e4acfab561fe593ad3f7a";
-
-    // when indexing IRM
-    ResponseEntity<String> response =
-        restTemplate.exchange(
-            indexEndpointUrl,
-            HttpMethod.PUT,
-            createIndexRequest("{ \"ext.extracted.text\" : \"" + contents + "\" }"),
-            String.class);
-
-    // then
-    assertThat(response.getStatusCode(), is(HttpStatus.OK));
+  @AfterEach
+  void afterEach() throws IOException {
+    storeMockWebServer.shutdown();
   }
 
-  @ParameterizedTest
-  @ValueSource(strings = {"", "{}", "{ \"\": \"text\"}"})
-  public void testStoringInvalidCst(String contents) throws IOException {
-    // given
-    final String indexEndpointUrl = INDEX_ENDPOINT_BASE_URL + "00067360b70e4acfab561fe593ad3f7a";
-    // when indexing IRM
-    ResponseEntity<String> response =
-        restTemplate.exchange(
-            indexEndpointUrl, HttpMethod.PUT, createIndexRequest(contents), String.class);
-
-    // then
-    assertThat(response.getStatusCode(), is(HttpStatus.BAD_REQUEST));
-  }
+  @Test
+  void testContextLoads() {}
 
   @Test
-  public void testStoreMetadataCstWhenSolrIsEmpty() throws Exception {
-    // given
-    final String queryKeyword = "Winterfell";
-    final String indexEndpointUrl = INDEX_ENDPOINT_BASE_URL + "00067360b70e4acfab561fe593ad3f7a";
-    final String irmLocation =
-        String.format(
-            "%s/%s/%s",
-            datasetRetrieveEndpoint, "00067360b70e4acfab561fe593ad3f7a", IRM_PATH_SEGMENT);
-
-    // when indexing IRM
-    restTemplate.put(
-        indexEndpointUrl,
-        createIndexRequest(
-            "{ \"ext.extracted.text\" : \""
-                + ("All the color had been leached from "
-                    + queryKeyword
-                    + " until only grey and white remained")
-                + " \" }"));
-
-    // then
-    final URIBuilder queryUriBuilder = new URIBuilder();
-    queryUriBuilder.setPath(QueryController.URL_TEMPLATE);
-    queryUriBuilder.setParameter(
-        "q", String.format("%s LIKE '%s'", CONTENTS_ATTRIBUTE, queryKeyword));
-    assertThat(
-        (List<String>) restTemplate.getForObject(queryUriBuilder.build(), List.class),
-        hasItem(irmLocation));
+  void testIndexWhenSolrIsEmpty() throws Exception {
```

Comment:
✏️ In another PR, these tests that specifically mention Solr and are testing Solr indirectly should be either removed or pushed down to a more Solr specific test.

Reply:
You're right. These tests should be moved to a lower level. But I'm sick of working on this PR. Let's address after this PR is merged.